Lesson2.ipynb

```diff
@@ -648,6 +589,580 @@
+    "\n",
+    "```python\n",
+    "# An empty list\n",
+    "myList = []\n",
+    "\n",
+    "# A list of integers\n",
+    "myList = [1,2,10,220]\n",
+    "\n",
+    "# A list of integers, floats, strings and booleans\n",
+    "myList = [1,\"2\",10.5, False]\n",
+    "\n",
+    "# A list of Lists! (also called nested lists)\n",
+    "myList = [[1,2], [3,4]]\n",
+    "```\n",
+    "\n",
+    "Remember when you learnt that a String is like a List of characters?\n",
+    "\n",
+    "So, let's see one more time how to access an element of the list and how to get the number of elements in a list:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Lists are index-based and the first element is at index 0:\n",
+    "shopping_list = ['Apple', 'Banana', 'Orange', \"Strawberry\", \"Mango\"]\n",
+    "\n",
+    "print(\"Accessing the first element of the list (Index 0):\", shopping_list[0])\n",
+    "\n",
+    "# You can get a sublist of items by using a range of indices separated by a colon.\n",
+    "# First index is inclusive, second index is exclusive\n",
+    "print(\"Get the elements from index 1 to 4:\", shopping_list[1:4])\n",
+    "\n",
+    "# Changing an element of the list\n",
+    "shopping_list[2] = \"Cherry\"\n",
+    "print(\"I've changed the third element (Index 2)\")\n",
+    "print(\"Here's my new shopping list:\", shopping_list)\n",
+    "\n",
+    "# We use the len() method to get the numbers of elements in a List\n",
+    "print(\"The Shopping list has\", len(shopping_list), \"items\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Operators\n",
+    "\n",
+    "##### Append\n",
+    "\n",
+    "The `append(x)` method adds the element `x` to the end of the list:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "myList = [1,20]\n",
+    "print(\"Before append():\", myList)\n",
+    "\n",
+    "myList.append(10)\n",
+    "print(\"After append():\", myList)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "To add multiple elements at once, you can use the `+` operator to combine two lists:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "myList = [1,2,3]\n",
+    "print(\"Before +:\", myList)\n",
+    "\n",
+    "myList = myList + [4,5,6]\n",
+    "print(\"After +:\", myList)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "##### Remove\n",
+    "\n",
+    "The `remove(x)` method removes the first item from the list whose value is `x`. It is an error if there is no such item:"
```

Comment:
"It is an error if there is no such item" may read better as "An error is returned if there is no such item"
Can you include a code example to remove a non-existent item, so students can see?